cmd/ctr/list.go

```diff
@@ -6,46 +6,94 @@ import (
 	"text/tabwriter"
 
 	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/api/services/execution"
+	tasktypes "github.com/containerd/containerd/api/types/task"
 	"github.com/urfave/cli"
 )
 
-var containersCommand = cli.Command{
-	Name: "containers",
+var taskListCommand = cli.Command{
+	Name:    "tasks",
+	Usage:   "manage tasks",
+	Aliases: []string{"t"},
+	Subcommands: []cli.Command{
+		{
+			Name:    "list",
+			Usage:   "list tasks",
+			Aliases: []string{"ls"},
+			Flags: []cli.Flag{
+				cli.BoolFlag{
+					Name:  "quiet, q",
+					Usage: "print only the task id & pid",
+				},
+			},
+			Action: taskListFn,
+		},
+	},
 }
 
-var listCommand = cli.Command{
-	Name:    "list",
-	Aliases: []string{"ls"},
-	Usage:   "list containers",
-	Flags: []cli.Flag{
-		cli.BoolFlag{
-			Name:  "quiet, q",
-			Usage: "print only the container id",
+var containerListCommand = cli.Command{
+	Name:    "containers",
+	Usage:   "manage containers (metadata)",
+	Aliases: []string{"c"},
+	Subcommands: []cli.Command{
+		{
+			Name:    "list",
+			Usage:   "list tasks",
+			Aliases: []string{"ls"},
+			Flags: []cli.Flag{
+				cli.BoolFlag{
+					Name:  "quiet, q",
+					Usage: "print only the container id",
+				},
+			},
+			Action: containerListFn,
 		},
 	},
-	Action: func(context *cli.Context) error {
-		var (
-			quiet       = context.Bool("quiet")
-			ctx, cancel = appContext(context)
-		)
-		defer cancel()
+}
 
-		client, err := newClient(context)
-		if err != nil {
-			return err
-		}
-		containers, err := client.Containers(ctx)
-		if err != nil {
-			return err
-		}
-		if quiet {
-			for _, c := range containers {
-				fmt.Println(c.ID())
+func taskListFn(context *cli.Context) error {
+	var (
+		quiet       = context.Bool("quiet")
+		ctx, cancel = appContext(context)
+	)
+	defer cancel()
+
+	client, err := newClient(context)
+	if err != nil {
+		return err
+	}
+	containers, err := client.Containers(ctx)
+	if err != nil {
+		return err
+	}
+
+	tasks := client.TaskService()
+
+	tasksResponse, err := tasks.List(ctx, &execution.ListRequest{})
+	if err != nil {
+		return err
+	}
+
+	// Join with tasks to get status.
+	tasksByContainerID := map[string]*tasktypes.Task{}
+	for _, task := range tasksResponse.Tasks {
+		tasksByContainerID[task.ContainerID] = task
+	}
+
+	if quiet {
+		for _, c := range containers {
+			task, ok := tasksByContainerID[c.ID()]
+			if ok {
+				fmt.Printf("%s\t%d\n", c.ID(), task.Pid)
+			} else {
+				//Since task is not running, PID is printed 0
+				fmt.Printf("%s\t%d\n", c.ID(), 0)
 			}
-			return nil
 		}
+	} else {
 
 		w := tabwriter.NewWriter(os.Stdout, 10, 1, 3, ' ', 0)
+		fmt.Fprintln(w, "\nTASK LIST")
```

Comment:
Please remove this. Only header should be present. You can change `ID` to `TASK` to make it more obvious.